frontend/src/queries/schema.ts

```diff
@@ -391,6 +392,18 @@ export interface SavedInsightNode extends Node, InsightVizNodeViewProps, DataTab
 
 // Insight viz node
 
+/** Chart specific rendering options.
+ * Use ChartRenderingMetadata for non-serializable values, e.g. onClick handlers
+ * @see ChartRenderingMetadata
+ * **/
+export interface VizSpecificOptions {
+    [InsightType.RETENTION]?: {
+        hideLineGraph?: boolean
+        hideSizeColumn?: boolean
+        useSmallLayout?: boolean
```

Comment:
This is the only property I don't think makes sense. 
We should do this with container queries given that it looks nicer regardless of the setting, it's more based on the container size anyways which applies no matter where it is rendered right?

Reply:
I don't 100% agree, you might want to use this with a large container to show a ton of data, e.g. if you had 54 columns to show weekly retention over a year. We could do something smart like try to figure out whether what the user has asked for fits inside the container, and use the small layout if it doesn't, but I don't want to overengineer this.

Reply:
Fair enough. Then lets go with it for now